render/fx_renderer/fx_pass.c

```diff
@@ -1015,6 +1026,219 @@ bool fx_render_pass_add_optimized_blur(struct fx_gles_render_pass *pass,
 	return !failed;
 }
 
+static void render_drop_shadow_blur_direction(struct fx_gles_render_pass *pass,
+		const struct fx_render_drop_shadow_options *fx_options,
+		bool is_horizontal) {
+	const struct wlr_render_texture_options *options = &fx_options->tex_options.base;
+	struct fx_renderer *renderer = pass->buffer->renderer;
+	struct fx_texture *texture = fx_get_texture(options->texture);
+	assert(texture->has_alpha);
```

Comment:
are we sure we want to make this assertion? This would put the onus on the compositor to check if the texture has alpha. In the case of a scene buffer without alpha, should we fall back to the box shadow instead?